src/core/lib/iomgr/tcp_generic_options.h

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_CORE_LIB_IOMGR_TCP_GENERIC_OPTIONS_H
+#define GRPC_CORE_LIB_IOMGR_TCP_GENERIC_OPTIONS_H
+
+#include <grpc/support/port_platform.h>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/string_view.h"
+
+#include <grpc/event_engine/endpoint_config.h>
+#include <grpc/impl/codegen/grpc_types.h>
+
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/iomgr/socket_mutator.h"
+#include "src/core/lib/resource_quota/api.h"
+
+typedef struct grpc_tcp_generic_options {
```

Comment:
I think we are probably going to want a slightly different options struct for each iomgr impl.  For example, the Windows iomgr impl doesn't support a socket mutator, so there's no reason for it to have a field in that struct.

I think the struct you have here is really the one we want for posix, and that should probably live in socket_utils_posix.h.